File: java/src/main/java/Series.java
import java.util.List;
import java.util.stream.IntStream;

class Series {
  private final String string;

  Series(final String string) {
    this.string = string;
  }

  List<String> slices(final int num) {
    if (num <= 0) throw new IllegalArgumentException("Slice size is too small.");
    if (num > string.length()) throw new IllegalArgumentException("Slice size is too big.");
    return IntStream.rangeClosed(0, string.length() - num)
                    .boxed()
                    .map(i -> string.substring(i, i + num))
                    .toList();
  }
}

File: scala/src/main/scala/Series.scala
import scala.annotation.tailrec

object Series {
  private def toDigitsList(str: String): List[Int] =
    str.map(_.toInt - '0').toList

  @tailrec
  def slices(length: Int, str: String, list: List[List[Int]] = List()): List[List[Int]] =
    if (str.length < length) list
    else slices(length, str.tail, list :+ toDigitsList(str.substring(0, length)))
}